typescript/testsuite/datastore/datastore_service_client/datastore_client.ts

```diff
@@ -0,0 +1,181 @@
+                if (axios.default.isAxiosError(exception)) {
+                    const exceptionAxiosErr: axios.AxiosError = exception as axios.AxiosError;
+                    if (exceptionAxiosErr.response && exceptionAxiosErr.response.status === httpStatusCode.StatusCodes.NOT_FOUND) {
+                        return ok(false);
+                    }
+                }
+                return err(exception as Error);
+            }
+            return err(new Error("Performing a get request threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+        
+        //NOTE: axios will throw an exception on a non-200 status code
+        return ok(true);
+    }
+    
+    /*
+    Gets the value for a given key from the datastore
+    */
+    public async get(key: string): Promise<Result<string, Error>> {
+        const url: string = this.getUrlForKey(key);
+        let resp: axios.AxiosResponse<any>;
+        try {
+            resp = await axios.default.get(url);
+        } catch(exception: any) {
+            // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+            // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+            if (exception && exception.stack && exception.message) {
+                return err(exception as Error);
+            }
+            return err(new Error("Performing a get request threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+        
+        const body: any = resp.data;
+        let bodyString: string = String(body);
+
+        return ok(bodyString);
+    }
+
+    /*
+    Puts a value for the given key into the datastore
+    */
+    public async upsert(key: string, value: string): Promise<Result<null, Error>> {
+        const url: string = this.getUrlForKey(key); 
+        let resp: axios.AxiosResponse<any>;
+        try {
+            resp = await axios.default.post(url, value);
+        } catch(exception: any) {
+            // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+            // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+            if (exception && exception.stack && exception.message) {
+                return err(exception as Error);
+            }
+            return err(new Error("Performing a post request threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+
+        return ok(null);
+    }
+
+    public getUrlForKey(key: string): string {
+        return "http://"+this.ipAddr+":"+this.port+"/"+KEY_ENDPOINT+"/"+key;
+    }
+
+    /*
+    Wait for healthy response
+    */
+    public async waitForHealthy(retries: number, retriesDelayMilliseconds: number): Promise<Result<null, Error>> {
```

Comment:
Also this is a `public` method - it should come before any `private` or `private static` methods